tests/components/fan/test_dyson.py

```diff
@@ -85,18 +133,168 @@ def test_setup_component_with_no_devices(self):
     def test_setup_component(self):
         """Test setup component with devices."""
         def _add_device(devices):
-            assert len(devices) == 1
+            assert len(devices) == 2
             assert devices[0].name == "Device_name"
 
         device_fan = _get_device_on()
+        device_purecool_fan = _get_dyson_purecool_device()
         device_non_fan = _get_device_off()
 
-        self.hass.data[dyson.DYSON_DEVICES] = [device_fan, device_non_fan]
+        self.hass.data[dyson.DYSON_DEVICES] = [device_fan,
+                                               device_purecool_fan,
+                                               device_non_fan]
         dyson.setup_platform(self.hass, None, _add_device)
 
-    @mock.patch('libpurecoollink.dyson.DysonAccount.devices',
+    @mock.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+    @mock.patch('libpurecool.dyson.DysonAccount.devices',
+                return_value=[_get_dyson_purecool_device()])
+    def test_purecool_service_set_night_mode(self,
```

Comment:
Please use standalone pytest test functions for new tests.

Reply:
This looks like new tests, so I would prefer to have them as standalone test functions.